vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/VariantExporter.java

```diff
@@ -66,39 +70,53 @@ public VariantExporter() {
         outputSampleNames = new HashSet<>();
     }
 
-    public List<VariantContext> export(VariantDBIterator iterator, Region region) {
+    public List<VariantContext> export(VariantWithSamplesAndAnnotationsService variantService, QueryParams queryParams, Region region) {
         List<VariantContext> variantsToExport = new ArrayList<>();
         failedVariants = 0;
 
         // region sequence contains the last exported region: we set it to null to get the new region sequence from cellbase if needed
 
-        while (iterator.hasNext()) {
-            Variant variant = iterator.next();
-            if (region.contains(variant.getChromosome(), variant.getStart())) {
-                try {
-                    VariantContext variantContext = variantToVariantContextConverter.transform(variant);
-                    variantsToExport.add(variantContext);
-                } catch (Exception e) {
-                    logger.warn("Variant {}:{}:{}>{} dump failed: {}", variant.getChromosome(), variant.getStart(),
-                                variant.getReference(),
-                                variant.getAlternate(), e.getMessage());
-                    failedVariants++;
+        try {
+            // TODO
+            List<VariantRepositoryFilter> filters = new FilterBuilder()
+                    .getVariantEntityRepositoryFilters(queryParams.getMaf(), queryParams.getPolyphenScore(),
+                                                       queryParams.getSiftScore(), queryParams.getStudies(), queryParams.getConsequenceType());
+
+            List<VariantWithSamplesAndAnnotation> variants = variantService.findByRegionsAndComplexFilters(
+                    Collections.singletonList(region), filters, null , null, null);
+
+            for (VariantWithSamplesAndAnnotation variant : variants) {
+                if (region.contains(variant.getChromosome(), variant.getStart())) {
+                    try {
+                        VariantContext variantContext = variantToVariantContextConverter.transform(variant);
+                        variantsToExport.add(variantContext);
+                    } catch (Exception e) {
+                        logger.warn("Variant {}:{}:{}>{} dump failed: {}", variant.getChromosome(), variant.getStart(),
+                                    variant.getReference(),
+                                    variant.getAlternate(), e.getMessage());
+                        failedVariants++;
+                    }
                 }
             }
+        } catch (AnnotationMetadataNotFoundException e) {
+            e.printStackTrace();
```

Comment:
Add some treatment for this error. Understand where and why this exception can be thrown and add a logger message if the execution can continue, or stop it if it cannot.